docs/examples/use_cases/pytorch/efficientnet/main.py

```diff
@@ -648,30 +653,56 @@ def main(args, model_args, model_arch):
         best_prec1,
     ) = prepare_for_training(args, model_args, model_arch)
 
-    train_loop(
-        trainer,
-        lr_policy,
-        train_loader,
-        train_loader_len,
-        val_loader,
-        logger,
-        start_epoch=start_epoch,
-        end_epoch=(
-            min((start_epoch + args.run_epochs), args.epochs)
-            if args.run_epochs != -1
-            else args.epochs
-        ),
-        early_stopping_patience=args.early_stopping_patience,
-        best_prec1=best_prec1,
-        prof=args.prof,
-        skip_training=args.evaluate,
-        skip_validation=args.training_only,
-        save_checkpoints=args.save_checkpoints and not args.evaluate,
-        checkpoint_dir=args.workspace,
-        checkpoint_filename=args.checkpoint_filename,
-        keep_last_n_checkpoints=args.gather_checkpoints,
-        topk=args.topk,
-    )
+    class DALIServerCtx:
```

Comment:
Another magic wrapper which makes me think that using Proxy is hard. How about expanding the documentation to demystify the magic that happens here?